pkg/runtime/wfengine/workflow.go

```diff
@@ -214,6 +217,70 @@ func (wf *workflowActor) getWorkflowMetadata(ctx context.Context, actorID string
 	return metadata, nil
 }
 
+// This method purges all the completed activity data from a workflow associated with the given actorID
+func (wf *workflowActor) purgeWorkflowState(ctx context.Context, actorID string) error {
+	state, exists, err := wf.loadInternalState(ctx, actorID)
+	if err != nil {
+		return err
+	} else if !exists {
+		return api.ErrInstanceNotFound
+	}
+
+	runtimeState := getRuntimeState(actorID, state)
+	if !runtimeState.IsCompleted() {
+		return api.ErrNotCompleted
+	}
+
+	// Loop through the history of the loaded workflow state and schedule purge calls for all the activities that have been completed
+
+	// RRL TODO: Looping through history and invoking each actor once. They are not in the same transaction. Eqch transaction happens as its own call
+	// We are deleting one by one in sequence and not in builk.
+	// In case of each actor returning an error, the entire function returns an error
+	// Everything before error is deleted, and erverything afyter is not.
+	for _, e := range state.History {
+		if ts := e.GetTaskScheduled(); ts != nil {
+			targetActorID := getActivityActorID(actorID, e.EventId)
+			eventData, err := backend.MarshalHistoryEvent(e)
```

Comment:
We shouldn't be including the event data in the purge request.

Reply:
This still needs to be addressed (though the problem has moved to a new line now)